src/extractors/positiveFlux.ts

```diff
@@ -1,26 +1,25 @@
 import { normalizeToOne } from "../utilities";
 
 export default function ({
-  complexSpectrum,
-  previousComplexSpectrum,
+  ampSpectrum,
+  previousAmpSpectrum,
 }: {
-  complexSpectrum: { real: number[]; imag: number[] };
-  previousComplexSpectrum: { real: number[]; imag: number[] };
+  ampSpectrum: Float32Array;
+  previousAmpSpectrum: Float32Array;
 }): number {
-  if (!previousComplexSpectrum) {
+  if (!previousAmpSpectrum) {
     return 0;
   }
 
-  const normalizedRealComponent = normalizeToOne(complexSpectrum.real);
-  const previousNormalizedRealComponent = normalizeToOne(
-    previousComplexSpectrum.real
-  );
+  const normalizedMagnitudeSpectrum = normalizeToOne(ampSpectrum);
+  const previousNormalizedMagnitudeSpectrum =
+    normalizeToOne(previousAmpSpectrum);
 
   let sf = 0;
-  for (let i = 0; i < normalizedRealComponent.length; i++) {
+  for (let i = 0; i < normalizedMagnitudeSpectrum.length; i++) {
     let x =
-      Math.abs(previousNormalizedRealComponent[i]) -
-      Math.abs(normalizedRealComponent[i]);
+      Math.abs(normalizedMagnitudeSpectrum[i]) -
+      Math.abs(previousNormalizedMagnitudeSpectrum[i]);
     sf += Math.pow(Math.max(x, 0), 2);
```

Comment:
Actually, why do we have both a `positiveFlux` and a `spectralFlux` implementations?

Reply:
I thought we could separate them as features rather than add another global flag about ignoring decreases in magnitude between buffers that gets passed down, but I'm open to either way. This way there's code duplication, but I'm not too worried about that.